packages/protocol/contracts/L1/libs/LibUtils.sol

```diff
@@ -105,20 +70,77 @@ library LibUtils {
     }
 
     function getStateVariables(TaikoData.State storage state)
-        internal
+        external
         view
-        returns (TaikoData.StateVariables memory)
+        returns (TaikoData.SlotA memory slotA, TaikoData.SlotB memory slotB)
     {
-        TaikoData.SlotA memory a = state.slotA;
-        TaikoData.SlotB memory b = state.slotB;
+        slotA = state.slotA;
+        slotB = state.slotB;
+    }
+
+    function getSyncedSnippet(
+        TaikoData.State storage state,
+        TaikoData.Config memory config,
+        uint64 blockId
+    )
+        external
+        view
+        returns (ICrossChainSync.Snippet memory)
+    {
+        uint64 _blockId =
+            blockId == 0 ? state.slotB.lastVerifiedBlockId : blockId;
+        uint64 slot = _blockId % config.blockRingBufferSize;
+
+        TaikoData.Block storage blk = state.blocks[slot];
+
+        if (blk.blockId != _blockId) revert L1_BLOCK_MISMATCH();
+        if (blk.verifiedTransitionId == 0) revert L1_TRANSITION_NOT_FOUND();
+
+        TaikoData.TransitionState storage transition =
+            state.transitions[slot][blk.verifiedTransitionId];
```

Comment:
All this code seems duplicated from `getVerifyingTransition`, seems like that function should simply be reused here.

Reply:
getVerifyingTransition and getVariables in LibUtils are no longer used, so I removed them.